core/editor/src/tools/mod.rs

```diff
@@ -37,6 +69,10 @@ impl ToolState {
 			mod_keys: self.mod_keys,
 		})
 	}
+
+	pub fn active_tool(&mut self) -> &mut Box<dyn Tool> {
+		self.tools.get_mut(&self.active_tool_type).unwrap()
```

Comment:
Could you either avoid the unwrap by  returning an option or add a comment on why it is safe?

Reply:
I expected that it will be guaranteed that the `tools` HashMap will have every tool in it, so I used `unwrap` here. I will change it to `Result` return type I guess.

Reply:
I think it might be better to use an option and do the error handling one leve above

Reply:
What your saying is true, but might cause some headache when adding tools and forgetting to initialize them. This is not as bad because we have stack traces but it is cleaner that way

Reply:
I meant how to handle the error on the level above, just unwrap it?

Reply:
This is a core library, we can't throw a JS exception here.

Reply:
That is true it would be nice though if our core libraries could not panic, you could just add an new error type and propagate it through